### DISCUSSION TITLE
How to use List.find?

### DISCUSSION THREAD

**Discussion by abhin4v** [chrisdone/hell#61] [Q&A] [answered]
Using List.find throws Zonk error:

```
hell: Zonk error: Ambiguous meta variable.
CallStack (from HasCallStack):
  error, called at src/Hell.hs:135:35 in main:Main
```

Here is a test program:

```
main = do
  let x = List.find (\a -> Text.eq a "x") ["x", "y"]
  Monad.return ()
```

I suspect this is because of the `b` type variables in its signature:

```
List.find :: forall a b . (a -> Bool) -> [a] -> Maybe a
```

**Comment by chrisdone** [comment]
Good catch! Thank you for letting me know! I should probably have a test suite make sure all polymorphic names are exercised. 

That’s an easy fix; just drop `b` from the sig in Hell.hs and it’ll work. Feel free to open a PR, else I’ll fix this when I’m back at keyboard, which might not be until Tuesday. On vacation at the moment.

**Comment by chrisdone** [comment, marked as answer]
I got a brief minute at keyboard, so I pushed the fix to drop the `b` variable to master at https://github.com/chrisdone/hell/commit/befd15d43eba011ac00d308da41dcd776f015105.
